cetmix_tower_server/models/cetmix_tower.py

```diff
@@ -145,13 +145,10 @@ def server_check_ssh_connection(self, server_reference, attempts=5, timeout=15):
         elif server.ssh_auth_mode == "k":
             ssh_params["ssh_key"] = server.ssh_key_id.sudo().secret_value
 
-        # Initialize SSH connection instance
-        ssh_connection = SSH(**ssh_params)
-
         # Try connecting multiple times
         for attempt in range(1, attempts + 1):
             try:
-                ssh_connection.connection()
+                ssh_connection = SSH(**ssh_params)
```

Comment:
I don't understand why you are creating a new `SSH` instance with the same parameters on every attempt inside the loop. 

If the `SSH` instance itself does not maintain any state related to the connection and only represents an object of the class, then creating a new instance on each attempt is redundant and unnecessary.

Reply:
Yes, you are right. Thank you for noticing that!